code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -263,32 +263,21 @@ func (msc *MinerSmartContract) payFees(t *transaction.Transaction,
 	b := balances.GetBlock()
 	if b.Round == gn.ViewChange {
 		if err := msc.SetMagicBlock(gn, balances); err != nil {
-			return "", common.NewErrorf("pay_fee",
+			return "", common.NewErrorf("pay_fees",
 				"can't set magic b round=%d viewChange=%d, %v",
 				b.Round, gn.ViewChange, err)
 		}
 	}
 
 	if t.ClientID != b.MinerID {
-		return "", common.NewError("pay_fee", "not block generator")
+		return "", common.NewError("pay_fees", "not block generator")
 	}
 
 	if b.Round <= gn.LastRound {
-		return "", common.NewError("pay_fee", "jumped back in time?")
+		return "", common.NewError("pay_fees", "jumped back in time?")
 	}
 
-	// the b generator
-	var mn *MinerNode
-	if mn, err = getMinerNode(b.MinerID, balances); err != nil {
-		return "", common.NewErrorf("pay_fee", "can't get generator '%s': %v",
-			b.MinerID, err)
-	}
-
-	logging.Logger.Debug("Pay fees, get miner id successfully",
-		zap.String("miner id", b.MinerID),
-		zap.Int64("round", b.Round),
-		zap.String("block", b.Hash))
-
+	var minerRewards, sharderRewards, minerFees, sharderFees currency.Coin
```

Comment:
I'm not a fan of such a definition, we better have variable defined before the block it is used